File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/data/converters/MapConverter.java
package com.enpassio.twowaydatabinding.data.converters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Map;

public class MapConverter {

    @TypeConverter
    public static String convertMapToJson(Map<String, Boolean> categories){
        Type categoryMap = new TypeToken<Map<String, Boolean>>(){}.getType();
        return new Gson().toJson(categories, categoryMap);
    }

    @TypeConverter
    public static Map<String, Boolean> convertJsonToMap(String jsonToConvert){
        Type categoryMap = new TypeToken<Map<String, Boolean>>(){}.getType();
        return new Gson().fromJson(jsonToConvert, categoryMap);
    }

}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/data/MapConverter.kt
package com.enpassion.twowaydatabindingkotlin.data

import android.arch.persistence.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken


object MapConverter {

    @TypeConverter
    fun convertMapToJson(categories: Map<String, Boolean>): String {
        val categoryMap = object : TypeToken<Map<String, Boolean>>() {

        }.type
        return Gson().toJson(categories, categoryMap)
    }

    @TypeConverter
    fun convertJsonToMap(jsonToConvert: String): Map<String, Boolean> {
        val categoryMap = object : TypeToken<Map<String, Boolean>>() {

        }.type
        return Gson().fromJson(jsonToConvert, categoryMap)
    }

}